app/src/main/java/com/example/hris/ui/fragments/profile/UpdateProfileFragment.kt

```diff
@@ -76,11 +74,11 @@ class UpdateProfileFragment : Fragment() {
         }
 
         viewModel.loadingDialogState.observe(viewLifecycleOwner) {
-            loadingDialog.apiCalling(it, childFragmentManager)
+            (activity as MainActivity).setLoadingDialog(it)
```

Comment:
Avoid referencing a fragment's parent activity as a specific class. Use an `interface` or the activity VM to communicate between fragments and their parent activity.